### PR TITLE
lib/model: Don't leak fd when truncate fails (fixes #4593)

### LINKED ISSUE #4593: Fd leak when temp file cannot be truncated
If the truncate call fails after we've opened a temp file we will return with an error, leaving the temp file fd open.

### PR DESCRIPTION
**Pull request by calmh**
Also attempt to handle this nicer by ignoring the truncate failure when
it doesn't matter, and recover by deleting the temp file when it does.

### CODE REVIEW

File: lib/model/sharedpullerstate.go [hunk 1]
```diff
@@ -158,8 +158,27 @@ func (s *sharedPullerState) tempFile() (io.WriterAt, error) {
...
+				// should come around with s.reused == 0.
+
+				fd.Close()
+
+				if err := s.fs.Remove(s.tempName); err != nil {
+					l.Debugln("failed to remove temporary file:", err)
+				}
+
+				s.failLocked("dst truncate", err)
+				return nil, err
```
Comment (AudriusButkevicius) [inline lib/model/sharedpullerstate.go]: This is confusing, it seems like the err from the if would shadow this, but it probably doesn't

Comment (imsodin) [reply lib/model/sharedpullerstate.go]: Maybe name the error in the "removal if" something like `remErr` to make it obvious.

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as c7522063b35c3e0656d335375d4f19dbeb9a82b3. Thanks, @calmh!

**Pull request closed without merging**